linera-execution/src/committee.rs

```diff
@@ -155,6 +155,40 @@ impl Committee {
         self.validity_threshold
     }
 
+    pub fn validators(&self) -> &BTreeMap<ValidatorName, ValidatorState> {
+        &self.validators
+    }
+
+    pub fn total_votes(&self) -> u64 {
+        self.total_votes
+    }
+
+    pub fn insert(&mut self, name: ValidatorName, new_state: ValidatorState) -> bool {
+        let mut new_total = self.total_votes;
+        if let Some(state) = self.validators.get(&name) {
+            if *state == new_state {
+                return false; // No change.
+            }
+            new_total -= state.votes;
+        }
+        if let Some(new_total) = new_total.checked_add(new_state.votes) {
+            self.total_votes = new_total;
+        } else {
+            return false;
+        };
+        self.validators.insert(name, new_state);
```

Comment:
Question: don't the `*_threshold`s have to be updated here as well?

Reply:
Ah, sorry, I was meaning to delete this method again! (I forgot to save. :facepalm:)
I pushed the fixed commit.